src/arch/x86/x86_64/src/lib.rs

```diff
@@ -6,9 +6,11 @@
 
 const PAGE_SIZE: usize = 4096;
 pub mod acpi;
+pub mod amdromehsmp;
 pub mod bzimage;
```

Comment:
The other thing we learned the hard way: 
arch is the root of the tree 
cpu pulls mods from arch
soc pulls mods from cpu
mainboard should for the most part import soc

anything else turns into a huge mess, as it did in tock, with everything reference everything.
So if we move amdromehsmp to soc/amd/rome/src/hsmp, this pub mod line should not be here.